Source/WebKit/UIProcess/API/glib/WebKitFeature.cpp

```diff
@@ -0,0 +1,150 @@
+    }
+    RELEASE_ASSERT_NOT_REACHED();
+}
+
+static inline WebKitFeatureCategory toFeatureCategory(API::FeatureCategory category)
+{
+    switch (category) {
+    case API::FeatureCategory::None:
+        return WEBKIT_FEATURE_CATEGORY_NONE;
+    case API::FeatureCategory::Animation:
+        return WEBKIT_FEATURE_CATEGORY_ANIMATION;
+    case API::FeatureCategory::CSS:
+        return WEBKIT_FEATURE_CATEGORY_CSS;
+    case API::FeatureCategory::DOM:
+        return WEBKIT_FEATURE_CATEGORY_DOM;
+    case API::FeatureCategory::HTML:
+        return WEBKIT_FEATURE_CATEGORY_HTML;
+    case API::FeatureCategory::Javascript:
+        return WEBKIT_FEATURE_CATEGORY_JAVASCRIPT;
+    case API::FeatureCategory::Media:
+        return WEBKIT_FEATURE_CATEGORY_MEDIA;
+    case API::FeatureCategory::Networking:
+        return WEBKIT_FEATURE_CATEGORY_NETWORK;
+    case API::FeatureCategory::Privacy:
+        return WEBKIT_FEATURE_CATEGORY_PRIVACY;
+    case API::FeatureCategory::Security:
+        return WEBKIT_FEATURE_CATEGORY_SECURITY;
+    }
+    RELEASE_ASSERT_NOT_REACHED();
+}
+
+struct _WebKitFeature {
+    _WebKitFeature(RefPtr<API::Feature>&& feature)
+        : feature(WTFMove(feature))
+    {
+    }
+
+    RefPtr<API::Feature> feature;
+    CString name { feature->name().utf8() };
+    CString details { feature->details().utf8() };
+    int referenceCount { 1 };
+};
+
+G_DEFINE_BOXED_TYPE(WebKitFeature, webkit_feature, webkit_feature_ref, webkit_feature_unref)
+
+WebKitFeature* webkitFeatureCreate(RefPtr<API::Feature>&& apiFeature)
+{
+    WebKitFeature* feature = static_cast<WebKitFeature*>(fastMalloc(sizeof(WebKitFeature)));
+    new (feature) WebKitFeature(WTFMove(apiFeature));
+    return feature;
+}
+
+API::Feature& webkitFeatureGetFeature(WebKitFeature* feature)
+{
+    ASSERT(feature);
+    return *feature->feature;
+}
+
+WebKitFeature* webkit_feature_ref(WebKitFeature* feature)
+{
+    g_atomic_int_inc(&feature->referenceCount);
+    return feature;
+}
+
+void webkit_feature_unref(WebKitFeature* feature)
+{
+    if (g_atomic_int_dec_and_test(&feature->referenceCount)) {
+        feature->~WebKitFeature();
+        fastFree(feature);
+    }
+}
+
+const char* webkit_feature_get_name(WebKitFeature* feature)
+{
+    g_return_val_if_fail(feature, nullptr);
+    return feature->name.data();
+}
+
+const char* webkit_feature_get_details(WebKitFeature* feature)
+{
+    g_return_val_if_fail(feature, nullptr);
+    return feature->details.data();
+}
+
+WebKitFeatureStatus webkit_feature_get_status(WebKitFeature* feature)
+{
+    g_return_val_if_fail(feature, WEBKIT_FEATURE_STATUS_EMBEDDER);
+    return toFeatureStatus(feature->feature->status());
+}
+
+WebKitFeatureCategory webkit_feature_get_category(WebKitFeature* feature)
+{
+    g_return_val_if_fail(feature, WEBKIT_FEATURE_CATEGORY_NONE);
+    return toFeatureCategory(feature->feature->category());
+}
+
+gboolean webkit_feature_get_default(WebKitFeature* feature)
```

Comment:
get_default_value?

Reply:
That does sound a bit clearer, I'll apply the change before pushing again 👌🏼